lib/model/tempname.go

```diff
@@ -20,6 +20,13 @@ type tempNamer struct {
 
 var defTempNamer tempNamer
 
+// (max filename length supported by FS) - len(".syncthing.") - len(".tmp")
+// Worst case is EncFS which according to man page supports filenames up to
+// "approximately" 3*(N-2)/4 characters where N is a limit of underlying FS.
+// If underlying FS is ext4 in practice it means that maximum filename length
+// is 188 characters. Minus len(".syncthing."), minus len(".tmp") gives:
+const maxFilenameLength = 173
```

Comment:
I'd make this

```go
const maxFilenameLength = 175 - len(".syncthing.") - len(".tmp")
```

to match observed reality. The "approximately" was very approximate indeed in my test case above.